apps/E2E/src/ButtonLegacy/pages/ButtonLegacyPageObject.ts

```diff
@@ -6,41 +6,17 @@ import {
   BUTTON_ON_PRESS_DEPRECATED,
 } from '../consts';
 import { BasePage, By } from '../../common/BasePage';
-
-/* This enum gives the spec file an EASY way to interact with SPECIFIC UI elements on the page.
- * The spec file should import this enum and use it when wanting to interact with different elements on the page. */
-export const enum ButtonSelector {
-  PrimaryButton, //this._primaryComponent
-}
-
 class ButtonLegacyPageObject extends BasePage {
   /******************************************************************/
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
-  async didOnClickCallbackFire(): Promise<boolean> {
-    const callbackText = await By(BUTTON_ON_PRESS_DEPRECATED);
-    await browser.waitUntil(async () => await callbackText.isDisplayed(), {
-      timeout: this.waitForUiEvent,
-      timeoutMsg: 'The OnClick callback did not fire.',
-      interval: 1000,
-    });
 
+  async didOnClickCallbackFire(errorMsg: string): Promise<boolean> {
+    const callbackText = await this._callbackText;
+    await this.waitForCondition(async () => await callbackText.isDisplayed(), errorMsg ?? 'The onClick callback did not fire.');
```

Comment:
No need for the `?? 'The onClick callback did not fire.'`.

Now that errorMsg is not optional, we know it'll be passed it